sql/parser/eval.go

```diff
@@ -259,6 +263,147 @@ func (d DString) String() string {
 	return StrVal(d).String()
 }
 
+// DDate is the date Datum.
+type DDate struct {
```

Comment:
The type alias doesn't give you access to the type methods. This is an embedded type within the struct that gives you access to the methods. It's weird that go doesn't support this!